internal/server/config/config.go

```diff
@@ -3,15 +3,19 @@ package config
 import (
 	"github.com/aquasecurity/trivy/internal/config"
 	"github.com/urfave/cli/v2"
+
+	"github.com/prometheus/client_golang/prometheus"
 )
 
 type Config struct {
 	config.GlobalConfig
 	config.DBConfig
 
-	Listen      string
-	Token       string
-	TokenHeader string
+	Listen          string
+	Token           string
+	TokenHeader     string
+	MetricsRegistry *prometheus.Registry
+	GaugeMetric     *prometheus.GaugeVec
```

Comment:
`Config` is for CLI options. Could you move it to the server package?

Reply:
So, @knqyf263 , actually, `MetricsRegistry` and `GaugeMetric` variables are supposed to be initialised exactly alongside the incoming CLI options and they will be the part of that configuration. 
Should I still go on and program the involvement of these variables in `internal/server/run.go` ? 

If so, I was thinking of creating another struct in `internal/server/run.go` like the following
```
type ServerConfig struct {
    config config.Config,
    MetricsRegistry *prometheus.Registry,
    GaugeMetric *prometheus.GaugeVec
}
```

and pass an object of type `ServerConfig` rather than `Config` in the last line of code which is, 
```return server.ListenAndServe(c, fsCache)``` 
(in the "c" argument)

Now, this will give errors because `server.ListenAndServe` function expects "c" to be of Type `Config`. So, to deal with this, I'll update `ListenAndServe` function in such a way that it expects "c" to be of type `ServerConfig` rather than `Config` and still deliver the output it already is delivering